react/week3/masi-todo-app/src/components/TodoListItem.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from "react";
+import { Checkbox, IconButton, Typography } from "@material-ui/core";
+import DeleteIcon from "@material-ui/icons/Delete";
+import { TextField } from "@material-ui/core";
+import { Add } from "@material-ui/icons";
+import PropTypes from 'prop-types';
+
+class TodoListItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { todoText: props.todo.title };
+
+  }
+  render() {
+    return (
+      this.props.todo.editing ? <div>
+        <TextField variant="outlined"
+          value={this.state.todoText}
+          onChange={(event) => {
+            this.setState({ todoText: event.target.value });
+          }}
```

Comment:
I see here that you are tracking the `todoText` by firstly getting it from the props and after editing it will be handled locally in the `TodoListItem`. 
The issue here is that by doing this you will get '2 sources of truth' for the `todo` text which are not in sync. In the parent component, `TodoList` you have only the first value and all the subsequent ones are inside the item. 
Please, adjust your code so the todo text is always coming from the parent component.

Reply:
very thoughtful point. thanks